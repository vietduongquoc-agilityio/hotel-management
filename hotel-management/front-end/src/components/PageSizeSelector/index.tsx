import React, { memo } from "react";
import { Select, Box } from "@chakra-ui/react";

interface PageSizeSelectorProps {
  pageSize: number;
  onPageSizeChange: (newSize: number) => void;
}

const PageSizeSelector: React.FC<PageSizeSelectorProps> = ({
  pageSize,
  onPageSizeChange,
}) => {
  const handlePageSizeChange = (
    event: React.ChangeEvent<HTMLSelectElement>
  ) => {
    const newSize = Number(event.target.value);
    onPageSizeChange(newSize);
  };

  return (
    <Box display="flex" textAlign="center" mt="3px" ml="10px">
      <Select
        width="55px"
        height="35px"
        mt="2px"
        cursor="pointer"
        value={pageSize}
        onChange={handlePageSizeChange}
        size="sm"
      >
        <option value={3}>3</option>
        <option value={5}>5</option>
        <option value={8}>8</option>
        <option value={10}>10</option>
        <option value={15}>15</option>
      </Select>
    </Box>
  );
};

export default memo(PageSizeSelector);
