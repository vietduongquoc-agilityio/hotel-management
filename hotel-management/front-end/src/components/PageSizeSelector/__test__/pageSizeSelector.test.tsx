import { render, screen, fireEvent } from "@testing-library/react";
import PageSizeSelector from "..";

describe("PageSizeSelector Component", () => {
  const mockOnPageSizeChange = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders correctly and matches snapshot", () => {
    const { asFragment } = render(
      <PageSizeSelector pageSize={10} onPageSizeChange={mockOnPageSizeChange} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the select dropdown with the correct default value", () => {
    render(<PageSizeSelector pageSize={10} onPageSizeChange={mockOnPageSizeChange} />);

    const selectElement = screen.getByRole("combobox");
    expect(selectElement).toBeInTheDocument();
    expect(selectElement).toHaveValue("10");
  });

  it("renders all the page size options", () => {
    render(<PageSizeSelector pageSize={10} onPageSizeChange={mockOnPageSizeChange} />);

    const options = screen.getAllByRole("option");
    expect(options).toHaveLength(4);

    expect(options[0]).toHaveValue("8");
    expect(options[0]).toHaveTextContent("8");

    expect(options[1]).toHaveValue("10");
    expect(options[1]).toHaveTextContent("10");

    expect(options[2]).toHaveValue("12");
    expect(options[2]).toHaveTextContent("12");

    expect(options[3]).toHaveValue("15");
    expect(options[3]).toHaveTextContent("15");
  });

  it("calls the onPageSizeChange callback with the correct value when a new page size is selected", () => {
    render(<PageSizeSelector pageSize={10} onPageSizeChange={mockOnPageSizeChange} />);

    const selectElement = screen.getByRole("combobox");
    fireEvent.change(selectElement, { target: { value: "12" } });

    expect(mockOnPageSizeChange).toHaveBeenCalledTimes(1);
    expect(mockOnPageSizeChange).toHaveBeenCalledWith(12);
  });

  it("updates the selected value when props change", () => {
    const { rerender } = render(
      <PageSizeSelector pageSize={10} onPageSizeChange={mockOnPageSizeChange} />
    );

    const selectElement = screen.getByRole("combobox");
    expect(selectElement).toHaveValue("10");

    rerender(<PageSizeSelector pageSize={15} onPageSizeChange={mockOnPageSizeChange} />);
    expect(selectElement).toHaveValue("15");
  });

  it("displays the select dropdown with the correct styling", () => {
    render(<PageSizeSelector pageSize={10} onPageSizeChange={mockOnPageSizeChange} />);
  });
});
