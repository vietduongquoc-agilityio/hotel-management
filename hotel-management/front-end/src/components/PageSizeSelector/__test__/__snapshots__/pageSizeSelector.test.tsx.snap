// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSizeSelector Component renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1cqyf5n"
  >
    <div
      class="chakra-select__wrapper css-4j1rev"
    >
      <select
        class="chakra-select css-1hlb958"
      >
        <option
          value="8"
        >
          8
        </option>
        <option
          value="10"
        >
          10
        </option>
        <option
          value="12"
        >
          12
        </option>
        <option
          value="15"
        >
          15
        </option>
      </select>
      <div
        class="chakra-select__icon-wrapper css-162mkon"
      >
        <svg
          aria-hidden="true"
          class="chakra-select__icon"
          focusable="false"
          role="presentation"
          style="width: 1em; height: 1em; color: currentColor;"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
